app/assets/javascripts/geolocation.js.erb

```diff
@@ -0,0 +1,31 @@
+document.addEventListener('DOMContentLoaded', function() {
```

Comment:
Thanks for figuring this all out! Looks kind of complicated, but reading through this I can mostly follow what is happening